yaas/blueprint.py

```diff
@@ -1,4 +1,138 @@
 # coding: utf-8
 
+from __future__ import absolute_import
+from __future__ import print_function
+
+import argparse
+import inspect
+import json
+import pprint
+import requests
+import sys
+
+from . import config
+
+
 def command(parser, args):
-    pass
+    """ Interact with Ambari blueprints. """
+    subparsers = parser.add_subparsers(dest='subcommand')
+    subcommands = {
+        'add': _add,
+        'list': _list,
+        'remove': _remove,
+        'show': _show,
+        }
+    for name, fn in subcommands.items():
+        subparsers.add_parser(name, help=inspect.getdoc(fn))
+    subargs, extra = parser.parse_known_args(args)
+    subcommands[subargs.subcommand](
+        subparsers.choices[subargs.subcommand],
+        extra)
+
+
+def _add(parser, args):
+    """ Add a blueprint to the Ambari server. """
+    parser.add_argument(
+        '--validate-topology',
+        default='true',
```

Comment:
* Not imposing `bool` here future-proofs better (e.g. Maybe `validate_topology=configurations_only` could be valid in the future).
* `argparse`+`bool` is a pain in the ass.
* The default is defined upstream which means if we are explicit here, a code change will be required if that default ever changes.
* Using `bool` requires us to solve a problem Ambari already solves: accepting `false` v. `False`, etc.